wave_vision/src/matcher/brute_force_matcher.cpp

```diff
@@ -5,12 +5,17 @@ namespace wave {
 
 // Default constructor. Struct may be default or user defined.
 BruteForceMatcher::BruteForceMatcher(const BFMatcherParams &config) {
+    bool cross_check;
+
     // Ensure parameters are valid
     this->checkConfiguration(config);
 
+    // Cross_check must be the opposite of use_knn
+    cross_check = !config.use_knn;
+
     // Create cv::BFMatcher object with the desired parameters
     this->brute_force_matcher =
-      cv::BFMatcher::create(config.norm_type, config.cross_check);
+      cv::BFMatcher::create(config.norm_type, cross_check);
 
     // Store configuration parameters within member struct
     this->current_config = config;
```

Comment:
Do this in initializer list.

Note since you may have a make a copy, the modern approach is to pass by value:
```cpp
BruteForceMatcher(BFMatcherParams config)
  : current_config{std::move(config)} {}
```

https://stackoverflow.com/questions/7592630/is-pass-by-value-a-reasonable-default-in-c11